### DISCUSSION TITLE
How to view targets and their properties when debugging CMake configuration

### DISCUSSION THREAD

**Discussion by itavero** [microsoft/vscode-cmake-tools#3448] [Q&A] [answered]
So far I'm really liking the new debug option introduced with CMake 3.27 that has been integrated into this extension as well.

While looking for some background information on it, I also came across a [blog post on how it is integrated in to _Visual Studio_](https://devblogs.microsoft.com/cppblog/cmake-debugger-allows-you-to-debug-your-cmake-scripts-and-more/) (so **not** VS Code).
They also highlight the ability to see the currently defined targets and their properties.

Is this feature also available in VS Code and, if so, where should I be able to find the targets and their properties?

**Comment by sinemakinci1** [comment, marked as answer]
Hi @itavero Thanks for reaching out! You are able to do the same in VS Code, see [our blog post on using in VS Code](https://devblogs.microsoft.com/cppblog/introducing-cmake-debugger-in-vs-code-debug-your-cmake-scripts-using-open-source-cmake-debugger/) Once you initiate a CMake debugging session, you should be able to see the state of your targets in the Variables pane in the debugger as shown in the GIF.

**Reply by itavero** [reply]
I see now. It appears that this only lists the targets from the current subdirectory, not all targets that are available. That's probably why I did not see it initially as the targets were defined in a subdirectory.

How can I see all targets that have been defined?

**Reply by sinemakinci1** [reply]
So, the best place to view all targets that have been defined is in the Project Outline in the CMake Tools sidebar. That being said, you won't be able to view the state of these variables from this view. Adding @gcampbell-msft as an FYI for targets viewable in the CMake debugger.
